tests/functionnal/Entity.php

```diff
@@ -517,4 +523,186 @@ public function testGetCustomCssTag(
+      $_SESSION['glpiactiveprofile']['interface'] = $interface;
+
+      // Set entity setting
+      $entity = getItemByTypeName("Entity", "_test_root_entity");
+      $update = $entity->update([
+         'id'                       => $entity->getID(),
+         'anonymize_support_agents' => $setting,
+      ]);
+      $this->boolean($update)->isTrue();
+
+      // // Set user nickname
+      $user = getItemByTypeName('User', 'tech');
+
+      if ($user_nick == "" && $user->fields['nickname'] == null) {
+         // Special case, glpi wont update null to "" so we need to set
+         // another value first
+         $update = $user->update([
+            'id'       => $user->getID(),
+            'nickname' => 'TMP',
+         ]);
+         $this->boolean($update)->isTrue();
+         $this->boolean($user->getFromDB($user->getID()))->isTrue();
+         $this->string($user->fields['nickname'])->isEqualTo('TMP');
+      }
+
+      $update = $user->update([
+         'id'       => $user->getID(),
+         'nickname' => $user_nick,
+      ]);
+      $this->boolean($update)->isTrue();
+      $this->boolean($user->getFromDB($user->getID()))->isTrue();
+      $this->string($user->fields['nickname'])->isEqualTo($user_nick);
+
+      // Build test ticket
+      $ticket = new Ticket();
+      $tickets_id = $ticket->add([
+         'name'             => 'test',
+         'content'          => 'test',
+         '_users_id_assign' => getItemByTypeName('User', 'tech', true),
+         'entities_id'      => $entity->getID(),
+      ]);
+      $this->integer($tickets_id)->isGreaterThan(0);
+      $this->boolean($ticket->getFromDB($tickets_id))->isTrue();
+
+      // Add followup to test ticket
+      $fup = new ITILFollowup();
+      $fup_id = $fup->add([
+         'content' => 'test',
+         'users_id' => getItemByTypeName('User', 'tech', true),
+         'users_id_editor' => getItemByTypeName('User', 'tech', true),
+         'itemtype' => 'Ticket',
+         'items_id' => $tickets_id,
+      ]);
+      $this->integer($fup_id)->isGreaterThan(0);
+
+      // Add solution to test ticket
+      $solution = new ITILSolution();
+      $solutions_id = $solution->add([
+         'content' => 'test',
+         'users_id' => getItemByTypeName('User', 'tech', true),
+         'users_id_editor' => getItemByTypeName('User', 'tech', true),
+         'itemtype' => 'Ticket',
+         'items_id' => $tickets_id,
+      ]);
+      $this->integer($solutions_id)->isGreaterThan(0);
+
+      // Case 1: test values recovered from CommonITILObject::showUsersAssociated()
+      ob_start();
+      $ticket->showUsersAssociated(CommonITILActor::ASSIGN, false, []);
+      $html = ob_get_clean();
+
+      foreach ($possible_values as $value) {
+         if ($value == $expected) {
+            $this->string($html)->contains($value);
+         } else {
+            $this->string($html)->notContains($value);
+         }
```

Comment:
I do not get why you do not check `$this->string($html)->contains($expected)`.

Reply:
I don't get what you are trying to say here.
Do you just mean doing  `$this->string($html)->contains($expected)` instead of `$this->string($html)->contains($value)` ?
It should be the same since we check `$value == $expected` just before.

More details about this:
`$possible_values` contains the 3 possible 'expected' usernames that will be supplied by the dataProvider (testAnonymizeSettingProvider).
For each test we check all the possibles values, one of them is expected and the others are not.

Reply:
ok